yt_dlp/extractor/youtube.py

```diff
@@ -941,54 +941,57 @@ def _parse_time_text(self, text):
-                # Downloading page may result in intermittent 5xx HTTP error
-                # Sometimes a 404 is also recieved. See: https://github.com/ytdl-org/youtube-dl/issues/28289
-                # We also want to catch all other network exceptions since errors in later pages can be troublesome
-                # See https://github.com/yt-dlp/yt-dlp/issues/507#issuecomment-880188210
-                if e.cause.status not in (403, 429):
-                    retry.error = e
-                    continue
-                return self._error_or_warning(e, fatal=fatal)
+                    first_bytes = e.cause.response.read(512)
+                    if not is_html(first_bytes):
+                        yt_error = try_get(
+                            self._parse_json(
+                                self._webpage_read_content(e.cause.response, None, item_id, prefix=first_bytes) or '{}', item_id, fatal=False),
+                            lambda x: x['error']['message'], str)
+                        if yt_error:
+                            self._report_alerts([('ERROR', yt_error)], fatal=False)
+                    # Downloading page may result in intermittent 5xx HTTP error
+                    # Sometimes a 404 is also recieved. See: https://github.com/ytdl-org/youtube-dl/issues/28289
+                    # We also want to catch all other network exceptions since errors in later pages can be troublesome
+                    # See https://github.com/yt-dlp/yt-dlp/issues/507#issuecomment-880188210
+                    if e.cause.status not in (403, 429):
+                        retry.error = e
+                        continue
+                    return self._error_or_warning(e, fatal=fatal)
 
-            try:
-                self._extract_and_report_alerts(response, only_once=True)
-            except ExtractorError as e:
-                # YouTube servers may return errors we want to retry on in a 200 OK response
-                # See: https://github.com/yt-dlp/yt-dlp/issues/839
-                if 'unknown error' in e.msg.lower():
-                    retry.error = e
+                try:
+                    self._extract_and_report_alerts(response, only_once=True)
+                except ExtractorError as e:
+                    # YouTube servers may return errors we want to retry on in a 200 OK response
+                    # See: https://github.com/yt-dlp/yt-dlp/issues/839
+                    if 'unknown error' in e.msg.lower():
+                        retry.error = e
+                        continue
+                    return self._error_or_warning(e, fatal=fatal)
+                # Youtube sometimes sends incomplete data
+                # See: https://github.com/ytdl-org/youtube-dl/issues/28194
+                if not traverse_obj(response, *variadic(check_get_keys)):
+                    icd_retry.error = ExtractorError('Incomplete data received', expected=True)
                     continue
```

Comment:
I think this should be `break` to tick the outer `RetryManager`?

Reply:
It does tick the outer one - I think because we don't set the inner retry.error that one exits and so the outer ticks over